index.js

```diff
@@ -1,34 +1,48 @@
-const ANNOTATION_EXECUTOR_TYPE = 'beta.screwdriver.cd/executor';
+const ANNOTATION_EXECUTOR_TYPE = 'beta.screwdriver.cd/executor'; // Key in annotations object that maps to an executor NPM module
 const Executor = require('screwdriver-executor-base');
 
 class ExecutorRouter extends Executor {
     /**
      * Constructs a router for different Executor strategies.
      * @method constructor
-     * @param  {Object}         options
-     * @param  {Array|String}   options.plugins                Array of plugins to load or a single plugin string
-     * @param  {String}         options.plugins[x].moduleName  Name of the executor NPM module to load
-     * @param  {String}         options.plugins[x].options     Configuration to construct the module with
+     * @param  {Object}         config                      Object with executor and ecosystem
+     * @param  {String}         [config.defaultPlugin]      Optional default executor
+     * @param  {Object}         [config.ecosystem]          Optional object with ecosystem values
+     * @param  {Array|Object}   config.executor             Array of executors to load or a single executor object
+     * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
+     * @param  {String}         config.executor[x].options  Configuration to construct the module with
      */
-    constructor(options = {}) {
+    constructor(config = {}) {
+        const ecosystem = config.ecosystem || {};
+        const executorConfig = config.executor;
+
+        if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
+            throw new Error('No executor config passed in.');
+        }
+
         super();
+        this.defaultExecutor = config.defaultPlugin;
 
-        const plugins = options.plugins;
-        const registrations = (Array.isArray(plugins)) ? plugins : [plugins];
+        executorConfig.forEach((plugin, index) => {
+            let ExecutorPlugin;
 
-        registrations.forEach((plugin, index) => {
-            // eslint-disable-next-line global-require, import/no-dynamic-require
-            const ExecutorPlugin = require(plugin.name);
+            try {
+                // eslint-disable-next-line global-require, import/no-dynamic-require
+                ExecutorPlugin = require(`screwdriver-executor-${plugin.name}`);
+            } catch (err) {
+                console.error(err);
+
+                return;
+            }
 
-            const executorPlugin = new ExecutorPlugin(plugin.options);
+            const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
+            const executorPlugin = new ExecutorPlugin(options);
 
             // Set the default executor
-            if (index === 0) {
-                this.default_executor = executorPlugin;
+            if (index === 0 && !config.defaultPlugin) {
+                this.defaultExecutor = executorPlugin;
```

Comment:
This module should probably throw if it does not have a valid defaultExecutor, after everything is initialized.